### PR TITLE
SRVKE-1520: Add transport-encryption documentation

### PR DESCRIPTION
**Pull request by pierDipi**
Testable after merging https://github.com/openshift-knative/eventing/pull/493

### CODE REVIEW

File: modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc
Comment (matzew) [inline modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:161]: does this only work with `strict`?

Comment (pierDipi) [reply modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:161]: no, permissive process is the same, only the config is different

File: modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc
Comment (ReToCode) [inline modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:30]: This seems rendered indented. Should this be a info block or something?

File: modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc
Comment (ReToCode) [inline modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:36]: Maybe also add that we do not recommend this for production use?

File: modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc
Comment (ReToCode) [inline modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:174]: Is there nothing that will run outside of eventing namespace that could need the bundle also?

**Review by ReToCode — COMMENTED** [review]
As a customer I probably also look for some information about rotation. We could delegate them to other docs (not sure if official cert-manager docs have something about this). At least we should probably have an info how eventing components behave when the CA is rotated (do I need to restart the components or not)? We could also hint to the additional bundle to trust multiple CAs during rotation.

File: modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc
Comment (creydr) [inline modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:26]:
```suggestion
* OpenShift CLI (`oc`) is installed.

* {serverlessoperatorname} is installed.

* {certmanageroperatorname} is installed.
```

nit. Maybe easier to read as it is a "prerequisite" :shrug:

File: modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc
Comment (creydr) [inline modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:84]:
```suggestion
<2> Secret name later used for the `ClusterIssuer` for Eventing
```

File: modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc
Comment (creydr) [inline modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:129]:
```suggestion
* *`disabled`* (this is equivalent to the current behavior)
```
(only not sure if this is supported in adoc)

File: modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc
Comment (creydr) [inline modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:132]:
```suggestion
* *`permissive`*
```

File: modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc
Comment (creydr) [inline modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:136]:
```suggestion
* *`strict`*
```

**Comment by pierDipi** [comment]
> As a customer I probably also look for some information about rotation. We could delegate them to other docs (not sure if official cert-manager docs have something about this). At least we should probably have an info how eventing components behave when the CA is rotated (do I need to restart the components or not)? We could also hint to the additional bundle to trust multiple CAs during rotation.

Good point, I would limit the text here on "info how eventing components behave when the CA is rotated" and leave to cert-manager the CA rotation part

File: modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc
Comment (pierDipi) [reply modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:36]: what would be our recommendation? use the same PKI as the configured OpenShift PKI or something else ?

File: modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc
Comment (pierDipi) [reply modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:30]: It is supposed to be a citation but info block sounds good

File: modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc
Comment (ReToCode) [reply modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:36]: Highly depends on what they (the customers company) already have. For a smooth rotation, for production, at least we should recommend to use the CA issuer (where they provide their own CA as a Secret).

Comment (pierDipi) [reply modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:36]: Right, I think I will stay vague here and just link cert-manager docs

**Comment by maschmid** [comment]
The "Configure additional CA bundles" part should probably make it clear that if I actually follow the "Setup a SelfSigned ClusterIssuer" part, I need to explictly create a trust-bundle with the `knative-eventing-selfsigned-ca` CA certificate  (and, should have an example steps showing how to do that)

File: modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc
Comment (pierDipi) [reply modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:26]: I copied it from here https://openshift-knative.github.io/docs/docs/latest/serverless/service-mesh/common-service-mesh-setup.html

image

**Comment by pierDipi** [comment]
Thanks for the reviews @ReToCode @creydr @maschmid, I've updated the PR with the feedback so it's ready for another pass

File: modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc
Comment (ReToCode) [reply modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:134]: I wanted to suggest the same. I think is supported, as above we have `transport-encryption`

File: modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc
Comment (ReToCode) [reply modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:179]: @pierDipi just to make sure, you do never deploy an Eventing component into another namespace than `knative-eventing` that would need the bundle?

File: modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc
Comment (ReToCode) [inline modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:233]:
```suggestion
1. Create a new CA certificate.
```

File: modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc
Comment (ReToCode) [inline modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:234]:
```suggestion
2. Add the public key of the new CA to the CA trust bundles as described in the <<configure_addition_ca_trust_bundles>> section. Make sure to also keep the public key of the existing CA.
```

File: modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc
Comment (ReToCode) [inline modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:235]:
```suggestion
3. Ensure that all clients have consumed the latest set of CA trust bundles. Knative Eventing components will automatically reload the changed CA bundles. If you have custom workload consuming trust bundles as well, make sure to reload/restart them accordingly.
```

Comment (ReToCode) [reply modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:235]: Not sure if this is correct. If not, you should probably add that people need to re-rollout components in `knative-eventing`?

File: modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc
Comment (ReToCode) [inline modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:238]: 6. As soon as the CA rotation is fully completed, you can remove the public key of the old CA from the trust bundle `ConfigMap`

File: modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc
Comment (pierDipi) [reply modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:179]: We deploy some but it's automatically handled for users, probably the only additional point I can add is how SinkBinding injects trust bundles into containers so that users can configure their HTTP clients to trust the CA

Comment (pierDipi) [reply modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc:179]: Added SB / workloads that connect to Eventing docs

**Review by ReToCode — APPROVED** [review]
/lgtm
/approve

/hold for other reviewers, feel free to unhold

**Comment by pierDipi** [comment]
@maschmid @matzew can we merge this one? please unhold if you agree

**Pull request merged**
